### PR TITLE
python3Packages.pyln-client: init at 0.8.0

### PR DESCRIPTION
**Pull request by zhenyavinogradov**
###### Motivation for this change

pyln-client is a client library for lightningd daemon: https://pypi.org/project/pyln-client/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kirelagin** [comment]
This obsoletes #68311, since `pylightning` is now kinda deprecated (they don’t state this anywhere explicitly, but what is happening is they are breaking their Python bindings into better organised modules, and at this time `pylightning` is just a compatibility shim around `pyln-client`). Plugins are being switched to `pyln-client` instead of `pylightning` too.

**Comment by kirelagin** [comment]
(Also, fwiw, works on macOS just fine.)

File: pkgs/development/python-modules/pyln-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  # the distribution lacks requirements.txt file, it should be empty
+  postPatch = ''
+    touch requirements.txt
+  '';
+
+  meta = with lib; {
+    description = "Client library for lightningd";
+    homepage = https://github.com/ElementsProject/lightning;
```
Comment (jonringer) [inline pkgs/development/python-modules/pyln-client/default.nix:21]: rfc 45
```suggestion
    homepage = "https://github.com/ElementsProject/lightning";
```

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -7328,6 +7328,8 @@ in {
 
   rxv     = callPackage ../development/python-modules/rxv     { };
 
+  pyln-client = callPackage ../development/python-modules/pyln-client { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:7331]: please attempt to order this

File: pkgs/development/python-modules/pyln-client/default.nix (hunk 1 above)
Comment (zhenyavinogradov) [reply pkgs/development/python-modules/pyln-client/default.nix]: Applied

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (zhenyavinogradov) [reply pkgs/top-level/python-packages.nix]: Ok, I tried

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by Artturin** [comment]
Rebase and reopen if still interested

**Pull request closed without merging**